File: front/src/services/storage/StorageService.js
import axios from "axios";


const STORAGE_BASE_URL = "http://localhost:8080/storages";

class StorageService {
    findById = async (storageId) => {
        return axios.get(`${STORAGE_BASE_URL}/${storageId}`);
    }

    findAll = async (pharmacyId, page = 1, size = 10) => {
        return axios.get(`http://localhost:8080/pharmacies/${pharmacyId}/storages?page=${page}&size=${size}`);
    }

    deleteById = async (storageId) => {
        return axios.delete(`${STORAGE_BASE_URL}/${storageId}`);
    }

    save = async (storage) => {
        return axios.post(`${STORAGE_BASE_URL}`, storage);
    }
}

export default new StorageService();

File: src/main/java/by/logvin/mip/service/StorageService.java
package by.logvin.mip.service;

import by.logvin.mip.model.entity.Storage;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface StorageService {
    Storage save(Storage storage);

    Storage findById(Long id);

    Page<Storage> findAll(Pageable pageable);

    Page<Storage> findAllStoragesByPharmacy(Long pharmacyId, Pageable pageable);

    void delete(Long id);
}
